engine-tests/src/test/java/org/terasology/engine/integrationenvironment/MTEExtensionTestWithPerMethodLifecycle.java

```diff
@@ -64,33 +60,31 @@ public static void createEntity(EntityManager entityManager, TestInfo testInfo)
     @Order(1)
     public void firstTestFindsThings() {
         List<EntityRef> entities = Lists.newArrayList(entityManager.getEntitiesWith(DummyComponent.class));
-        // There should be one entity, created by the @BeforeAll method
+        // There should be one entity, created by the @BeforeEach method
         assertEquals(1, entities.size());
 
         DummyComponent component = entities.get(0).getComponent(DummyComponent.class);
-        assertTrue(component.eventReceived);
+        assertThat(component.eventReceived).isTrue();
+        assertThat(seenNames).isEmpty();
 
         // Remember that a test has seen this one.
-        assertNotNull(component.name);
-        assertFalse(seenNames.containsKey(component.name));
-        seenNames.put(component.name, 1);
+        assertThat(component.name).isNotNull();
+        assertThat(seenNames).isEmpty();
+        seenNames.add(component.name);
     }
 
     @Test
     @Order(2)
-    public void thingsStillExistForSecondTest() {
+    public void thingsDoNotPolluteSecondTest() {
         List<EntityRef> entities = Lists.newArrayList(entityManager.getEntitiesWith(DummyComponent.class));
-        // There should be one entity, created by the @BeforeAll method
+        // There should be one entity, created by the @BeforeEach method for this one test
         assertEquals(1, entities.size());
 
-        // Make sure that this is the same one that the first test saw.
+        // FIXME: what to assert here does this make sense or should we drop the whole thing
+        //     and say it's covered by LifecyclePerMethodInjectionTest?
         DummyComponent component = entities.get(0).getComponent(DummyComponent.class);
-        assertTrue(component.eventReceived);
-        assertNotNull(component.name);
-        assertTrue(seenNames.containsKey(component.name), () ->
-                String.format("This is not the same entity as seen in the first test!%n"
-                        + "Current entity: %s%n"
-                        + "Previously seen: %s",
-                        component.name, seenNames.keySet()));
+        assertThat(component.eventReceived).isTrue();
+        assertThat(component.name).isNotNull();
+        assertThat(seenNames).isEmpty();
```

Comment:
This currently fails, because `seenNames` is a `static` class member and thus carried over even on `PER_METHOD` runs, it seems.